### PR TITLE
[unix socket] Add unix socket connector

### PR DESCRIPTION
**Pull request by vyzigold**
This adds unix socket connector to connectors.

### CODE REVIEW

File: connector/unix-socket.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &connector, nil
+}
+
+func (connector *UnixSocketConnector) connectSingleSocket(info *socketInfo, isIn bool) error {
+	var err error
+	if isIn {
+		os.Remove(info.Address.Name)
```
Comment (paramite) [inline connector/unix-socket.go:109]: Can we make this configurable? I can imagine use case, where we want to listen on socket created by other entity. Same is for output socket. Are we really sure that we will always write to already existing sockets?

File: connector/unix-socket.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+					})
+					connector.logger.Debug("Error while trying to read from unix socket.")
+					continue
+				}
+				msg := string(connector.msgBuffer[:n])
+				outchan <- msg
+				connector.logger.Metadata(map[string]interface{}{
+					"message": msg,
+				})
+				connector.logger.Debug("Recieved a message.")
```
Comment (paramite) [inline connector/unix-socket.go:163]: Received

File: connector/unix-socket.go (hunk 1 above)
Comment (vyzigold) [reply connector/unix-socket.go:109]: I tried this yesterday and I just couldn't make it work. 

If I remember it correctly, I tried the following:
1. In goroutine A I created a socket using the ListenUnixgram
2. In goroutine B I used the dial Unix on that socket.
3. In goroutine A I sent a message
4. In goroutine B I tried to read the message.

It looked like the message was sent, but goroutine B didn't receive anything.

**Comment by vyzigold** [comment]
I'll implement the your suggestions tomorrow.

**Pull request merged**